aya/src/maps/sock/sock_map.rs

```diff
@@ -108,26 +102,13 @@ impl<T: Deref<Target = Map> + DerefMut<Target = Map>> SockMap<T> {
                 io_error,
             })
     }
-}
-
-impl<T: Deref<Target = Map> + DerefMut<Target = Map>> SocketMap for SockMap<T> {
-    fn fd_or_err(&self) -> Result<RawFd, MapError> {
-        self.inner.fd_or_err()
-    }
-}
-
-impl TryFrom<MapRef> for SockMap<MapRef> {
-    type Error = MapError;
-
-    fn try_from(a: MapRef) -> Result<SockMap<MapRef>, MapError> {
-        SockMap::new(a)
-    }
-}
 
-impl TryFrom<MapRefMut> for SockMap<MapRefMut> {
-    type Error = MapError;
-
-    fn try_from(a: MapRefMut) -> Result<SockMap<MapRefMut>, MapError> {
-        SockMap::new(a)
+    fn check_bounds(&mut self, index: u32) -> Result<(), MapError> {
```

Comment:
I'm wondering if we could somehow unify it with `array::check_bounds` to not duplicate it here.

Reply:
Sure I could add this to the top level maps/mod.rs as a helper function

Reply:
Yeah it looks like I can make this functionality more shared/generic let me do that

Reply:
moved all of the shared code for bound checking and k/v size checks to maps/mod.rs :)